lang/LangPrimSource/SC_Ableton_Link.cpp

```diff
@@ -0,0 +1,195 @@
+		SetObject(g->sp, mTempoClockObj);
+		++g->sp;
+		SetFloat(g->sp, mTempo);
+		++g->sp;
+		SetFloat(g->sp, beats);
+		++g->sp;
+		SetFloat(g->sp, secs);
+		++g->sp;
+		SetObject(g->sp, mTempoClockObj);
+		runInterpreter(g, s_LinkTempoChanged, 5);
+		g->canCallOS = false;
+		gLangMutex.unlock();
+	});
+
+	auto timeline = mLink.captureAppTimeline();
+	auto linkTime = hrToLinkTime(inBaseSeconds);
+	timeline.requestBeatAtTime(inBaseBeats, linkTime, mQuantum);
+	mLink.commitAppTimeline(timeline);
+}
+
+void LinkClock::SetAll(double inTempo, double inBeats, double inSeconds)
+{
+
+	auto timeline = mLink.captureAppTimeline();
+	auto linkTime = hrToLinkTime(inSeconds);
+	timeline.setTempo(inTempo * 60., linkTime);
+	timeline.requestBeatAtTime(inBeats,linkTime, mQuantum);
+
+	mTempo = inTempo;
+	mBeatDur = 1. / inTempo;
+
+	mLink.commitAppTimeline(timeline);
+	mCondition.notify_one();
+}
+
+void LinkClock::SetTempoAtBeat(double inTempo, double inBeats)
+{
+	auto timeline = mLink.captureAppTimeline();
+	auto time = timeline.timeAtBeat(inBeats, mQuantum);
+	timeline.setTempo(inTempo*60., time);
+	mLink.commitAppTimeline(timeline);
+}
+
+void LinkClock::SetTempoAtTime(double inTempo, double inSeconds)
+{
+	auto timeline = mLink.captureAppTimeline();
+	timeline.setTempo(inTempo*60., hrToLinkTime(inSeconds));
+	mLink.commitAppTimeline(timeline);
+}
+
+void LinkClock::SetQuantum(double quantum)
+{
+	mQuantum = quantum;
+	mCondition.notify_one();
+}
+
+//Primitives
+int prLinkClock_NumPeers(struct VMGlobals *g, int numArgsPushed);
+int prLinkClock_NumPeers(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *a = g->sp;
+	LinkClock *clock = (LinkClock*)slotRawPtr(&slotRawObject(a)->slots[1]);
+
+	SetInt(a, clock->NumPeers());
+
+	return errNone;
+}
+
+
+int prLinkClock_SetQuantum(struct VMGlobals *g, int numArgsPushed);
+int prLinkClock_SetQuantum(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *a = g->sp - 1;
+	PyrSlot *b = g->sp;
+	LinkClock *clock = (LinkClock*)slotRawPtr(&slotRawObject(a)->slots[1]);
+
+	double quantum;
+	int err = slotDoubleVal(b, &quantum);
+	if(err) return errFailed;
+
+	clock->SetQuantum(quantum);
+
+	return errNone;
+}
+
+void initLinkPrimitives()
+{
+	int base, index=0;
+
+	base = nextPrimitiveIndex();
+
+	s_LinkTempoChanged = getsym("prTempoChanged");
+
+	definePrimitive(base, index++, "_LinkClock_New", prClock_New<LinkClock>, 4, 0);
+	definePrimitive(base, index++, "_LinkClock_SetBeats", prClock_SetBeats<LinkClock>, 2, 0);
+	definePrimitive(base, index++, "_LinkClock_SetTempoAtBeat", prClock_SetTempoAtBeat<LinkClock>, 3, 0);
+	definePrimitive(base, index++, "_LinkClock_SetTempoAtTime", prClock_SetTempoAtTime<LinkClock>, 3, 0);
+	definePrimitive(base, index++, "_LinkClock_SetAll", prClock_SetAll<LinkClock>, 4, 0);
+	definePrimitive(base, index++, "_LinkClock_NumPeers", prLinkClock_NumPeers, 1, 0);
+	definePrimitive(base, index++, "_LinkClock_SetQuantum", prLinkClock_SetQuantum, 2, 0);
+}
+
+#else
+void initLinkPrimitives()
+{
+}
```

Comment:
IIUC this shouldn't be here, as you protect the one call to `initLinkPrimitives()` with the same guard. plus, it's nicer to have a build failure when you try to call a function you shouldn't :)